scipy/stats/_hypotests.py

```diff
@@ -132,6 +134,227 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+
+    This performs a test of the goodness of fit of a cumulative distribution
+    function F compared to the empirical distribution function F_n of
+    observed random variates X_1, ..., X_n that are assumed to be independent
+    and identically distributed ([1]_). The null hypothesis is that the X_i
+    have cumulative distribution F.
+
+    Parameters
+    ----------
+    rvs : array_like
+        The observed values of the random variables, it should be a 1-D array.
+    cdf : str or callable
+        If a string, it should be the name of a distribution in `scipy.stats`.
+        If a callable, that callable is used to calculate the cdf.
+        ``cdf(x, *args) -> float``, it should be a cdf.
+    args : tuple, optional
+        Distribution parameters. These are assumed to be known, see Notes.
+
+    Returns
+    -------
+    statistic : float
+        Cramer-von Mises statistic.
+    pvalue :  float
+        The p-value.
+
+    See Also
+    --------
+    kstest
+
+    Notes
+    -----
+    The p-value relies on the approximation given by equation 1.8 in [2]_.
+    It is important to keep in mind that the p-value is only accurate if
+    one tests a simple hypothesis, i.e. the parameters of the reference
+    distribution are known. If the parameters are estimated from the data
+    (composite hypothesis), the computed p-value is not reliable.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Cramer-von_Mises_criterion
+    .. [2] Csorgo, S. and Faraway, J. (1996). The Exact and Asymptotic
+           Distribution of Cramer-von Mises Statistics. Journal of the
+           Royal Statistical Society, pp. 221-234.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Test the assumption of normality. In this example, one accepts
+    the assumption of a normal distribution at the significance level
+    of alpha=0.05.
+
+    >>> np.random.seed(626)
+    >>> x = stats.norm.rvs(size=500)
+    >>> w, p = stats.cvm_test(x, 'norm')
+    >>> w, p
+    (0.06342154705518796, 0.792680516270629)
+
+    We can also use `args` to check whether the data is drawn from a
+    shifted normal distribution, assuming a shift of 2 when the
+    true shift is 2.1. In that case, the null hypothesis is rejected
+    at the significance level of alpha=0.05.
+
+    >>> y = x + 2.1
+    >>> w, p = stats.cvm_test(y, 'norm', args=(2,))
+    >>> w, p
+    (0.4798693195559657, 0.044782228803623814)
+
+    Using a callable argument for cdf, this is equivalent to the
+    following:
+
+    >>> cdf = stats.norm.cdf
+    >>> stats.cvm_test(y, cdf, args=(2,))
+
+    """
+    if isinstance(cdf, str):
+        cdf = getattr(distributions, cdf).cdf
+
+    vals = np.sort(np.asarray(rvs))
+
+    if vals.size <= 1:
+        raise ValueError('The sample must contain at least two observations.')
+    if vals.ndim > 1:
+        raise ValueError('The sample must be one-dimensional.')
+
+    n = len(vals)
+    cdfvals = cdf(vals, *args)
```

Comment:
Maybe put this in a `try` `except`, giving a helpful message upon exception, to validate `cdf`.

Reply:
And test the other case:
![image](https://user-images.githubusercontent.com/6570539/90555040-982b6900-e14b-11ea-8b63-ac87249eea8c.png)

Reply:
even without error handling, the error is pretty easy to understand:

```
getattr(stats.distributions, "xyz").cdf
AttributeError: module 'scipy.stats.distributions' has no attribute 'xyz'
```

do you think this is enough? (it is aligned to `_parse_kstest_args` in stats.py)

Reply:
Oops, I don't have a problem with the line `cdf = getattr(distributions, cdf).cdf`. I think it's fine if the user passes in an invalid string because, as you say, the error they will get is pretty easy to understand.

Let me try again.

> And test the other case

I mean that currently codecov reports that in the existing tests, `if isinstance(cdf, str)` is never false; there are no tests for what happens if a callable is passed in for `cdf`;  I'm suggesting we should add a test like you have in the docstring:
```
cdf = stats.norm.cdf
stats.cvm_test(y, cdf, args=(2,))
```
although I can understand if we think the doctest itself is enough.

> Maybe put this in a `try` `except`

I was saying that if the user passes in something other than a string for `cdf`, but it is not actually a callable that represents a cumulative distribution function, they might get an error at this line. In that case, perhaps they could use a reminder about what `cdf` is supposed to be rather than whatever error executing `cdf(vals, *args)` happens to cause. One way to do this is to `try` `cdf(vals, *args)`, catch any exception that occurs, and re-raise it with an additional message about what `cdf` is supposed to be.

Neither of these suggestions are blocking, though, if there's a good reason not to implement them.

Reply:
I added a new test case, TBD if we want to introduce exception handling